src/libraries/Microsoft.PowerFx.Core/UtilityDataStructures/LazyList.cs

```diff
@@ -14,7 +14,7 @@ namespace Microsoft.PowerFx.Core.UtilityDataStructures
     /// the operation without the creation of many intermediate large
     /// memory blocks.
     /// </summary>
-    internal class LazyList<T> : IEnumerable<T>
+    public class LazyList<T> : IEnumerable<T>
```

Comment:
>public [](http://example.com/codeflow?start=4&length=6)

I'm skeptical of the value of this class... wondering if we can avoid making it public.